src/bin/tikv-server.rs

```diff
@@ -589,6 +595,42 @@ fn get_store_path(matches: &Matches, config: &toml::Value) -> String {
     format!("{}", absolute_path.display())
 }
 
+fn start_server<T, S>(mut server: Server<T, S>, mut el: EventLoop<Server<T, S>>)
+    where T: RaftStoreRouter,
+          S: StoreAddrResolver + Send + 'static
+{
+    let ch = server.get_sendch();
+    let h = thread::Builder::new()
+        .name("tikv-server".to_owned())
+        .spawn(move || {
+            server.run(&mut el).unwrap();
+        })
+        .unwrap();
+    handle_signal(ch);
+    h.join().unwrap();
+}
+
+#[cfg(unix)]
+fn handle_signal(ch: SendCh<Msg>) {
+    use signal::trap::Trap;
+    use nix::sys::signal::{SIGTERM, SIGINT};
+    let trap = Trap::trap(&[SIGTERM, SIGINT]);
```

Comment:
Docs: https://tailhook.github.io/signal/signal/trap/index.html
Examples: https://github.com/tailhook/signal/tree/master/examples
It use `sigwait` internally, see also http://man7.org/linux/man-pages/man3/sigwait.3.html